src/main/java/dev/mja00/villagerLobotomizer/utils/StringUtils.java

```diff
@@ -0,0 +1,39 @@
+package dev.mja00.villagerLobotomizer.utils;
+
+import java.util.Locale;
+
+public class StringUtils {
+    /**
+     * Compares two semantic version strings (e.g., "1.2.3").
+     *
+     * @return -1 if version1 < version2, 0 if equal, 1 if version1 > version2
+     */
+    public static int compareSemVer(String version1, String version2) {
```

Comment:
Should add a null/empty string sanity check here. 
Also, since we don't want to have them be null we should notate it with `@NotNull` (from Jetbrains' annotation package) that way most IDEs will yell. 


```suggestion
    public static int compareSemVer(@NotNull String version1, @NotNull String version2) {
```

Reply:
kk